### PR TITLE
Wip/hughsie/ccgx dmc fuzz

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/ccgx/fu-ccgx-dmc-firmware.c [hunk 1]
```diff
@@ -257,81 +254,93 @@ fu_ccgx_dmc_firmware_parse (FuFirmware *firmware,
...
+	if (!fu_memcpy_safe ((guint8 *) fwct_buf, hdr_size, 0x0,	/* dst */
+			     buf, bufsz, 0,				/* src */
+			     hdr_size, error)) {
+		g_prefix_error (error, "failed to read fwct: ");
 		return FALSE;
+	}
 
 	/* create fwct binary */
-	self->fwct_blob = g_bytes_new (fwct_buf, self->fwct_info.size);
+	self->fwct_blob = g_bytes_new (fwct_buf, hdr_size);
```
Comment (hughsie) [inline plugins/ccgx/fu-ccgx-dmc-firmware.c:317]: Bytes_new_offset

**Comment by chlee75** [comment]
@hughsie 

Hi. Richard.

I come back to work today due to holiday in Korea.
Code maintenance has been assigned to Dudley who is a new member in Infineon.
I will let you know his account in github so that he could review your changes.

Thanks.
Ryan.

**Comment by hughsie** [comment]
>  Code maintenance has been assigned to Dudley who is a new member in Infineon.

Should I leave your access or are you now working on other things?

> I will let you know his account in github so that he could review your changes.

Thanks.

**Comment by chlee75** [comment]
@hughsie 

> Should I leave your access or are you now working on other things?
 My job is firmware development in Asia and I have been temporarily holding LVFS.
 So I have been taking over my LVFS support to Dudley.
 Please keep my access until Dudley take LVFS completely.

Thanks.

**Pull request merged**